opensrp-app/src/main/java/org/smartregister/repository/P2PSenderTransferDao.java

```diff
@@ -28,37 +30,51 @@
     @Nullable
     @Override
     public TreeSet<DataType> getDataTypes() {
-        return (TreeSet<DataType>) dataTypes.clone();
+        TreeSet<DataType> dataTypeTreeSet = new TreeSet<>();
+        int start = dataTypes.size() + 1;
+        P2POptions p2POptions = CoreLibrary.getInstance().getP2POptions();
+        if (p2POptions != null && !ArrayUtils.isEmpty(p2POptions.getLocationsFilter())) {
+            for (String location : p2POptions.getLocationsFilter()) {
+                for (DataType dataType : dataTypes) {
+                    dataTypeTreeSet.add(new DataType(dataType.getName() + ":" + location, dataType.getType(), start + dataTypeTreeSet.size()));
+                }
+            }
+        }
+        TreeSet<DataType> dataTypesClone = new TreeSet<>(dataTypes);
+        dataTypesClone.addAll(dataTypeTreeSet);
+        return dataTypesClone;
     }
 
     @Nullable
     @Override
     public JsonData getJsonData(@NonNull DataType dataType, long lastRecordId, int batchSize) {
-        if (dataType.getName().equals(event.getName())) {
+        String[] dataTypeParams = dataType.getName().split(":");
```

Comment:
We can also depend on the presence of the location filter here to decide if we should extract the locationId